CONTRIBUTING.md

```diff
@@ -0,0 +1,41 @@
+# Contribution Guidelines
+
+netlink accepts contributions via GitHub pull request(PR). This document outlines some of the conventions on how to contribute
+code to the project.
+
+## General
+
+* One commit per PR.
```

Comment:
I really have a big issue with this; yes, a PR should not have "fix-up" commits, but I don't see a good reasons for making a single-commit-per-PR a requirement; also see my comment on https://github.com/vishvananda/netlink/pull/1048#issuecomment-2774966310

Commits should be logically grouped (and each commit in a PR should build), but a pull request may contain multiple commits to make them useful.

Reply:
As the creator of the initial policy, the thing I care about is making sure the history is linear and every commit can build properly. The easiest way to enforce this is to disallow merge commits and require one commit per pr. I think a series of commits that can be rebased cleanly onto main is also fine although I find that is a bit harder for newer contributors to do well.

Reply:
My goal with this PR is to reflect the current policies defined by the maintainers of the repo so we can have a baseline.

personally i usually do a merge commit or squash and merge in other GH repos where im a maintainer.

TBH i see far too often PRs with multiple commits that have really no business getting merged as is.

e.g
a PRs that end up having something of the sort of "(each bullet is a commit):

- forgot to cover some scenario via test
- fix that lint issue
- address review comments
- feat: initial implementaion

Reply:
> As the creator of the initial policy, the thing I care about is making sure the history is linear and every commit can build properly. The easiest way to enforce this is to disallow merge commits and require one commit per pr. I think a series of commits that can be rebased cleanly onto main is also fine although I find that is a bit harder for newer contributors to do well.

Exactly. This is the best way to guarantee each commit is self-contained. CI passes. It can be cleanly ported to other branches. It forces the author to push a finalized, tested change.

@thaJeztah In the past nine years, I very rarely saw a PR with multiple commits here. It's just a non-issue to discuss about, IMO.